c++/src/kj/compat/http-test.c++

```diff
@@ -6188,6 +6213,91 @@ KJ_TEST("CONNECT Server cancels read w/client") {
   listenTask.wait(waitScope);
 }
 
+class TestConnectResponseImpl final: public kj::HttpService::ConnectResponse, public kj::Refcounted {
+public:
+  TestConnectResponseImpl(kj::Own<kj::PromiseFulfiller<kj::HttpClient::ConnectResponse>> fulfiller)
+      : fulfiller(kj::mv(fulfiller)) {}
+
+  void setPromise(kj::Promise<void> promise) {
+    task = promise.eagerlyEvaluate([this](kj::Exception&& exception) {
+      if (fulfiller->isWaiting()) {
+        fulfiller->reject(kj::mv(exception));
+      } else {
+        kj::throwRecoverableException(kj::mv(exception));
+      }
+    });
+  }
+
+  kj::Own<kj::AsyncIoStream> accept(
+      uint statusCode, kj::StringPtr statusText, const kj::HttpHeaders& headers) override {
+    KJ_REQUIRE(statusCode >= 200 && statusCode < 300, "the statusCode must be 2xx for accept");
+
+    auto headersCopy = kj::heap(headers.clone());
```

Comment:
Does this need to be attached anywhere?

Reply:
I based on this code which doesn't attach it: https://github.com/capnproto/capnproto/blob/master/c++/src/kj/compat/http.c++#L6185. 

Should both be changed?

Reply:
Hmm, I suspect so. FYI @jasnell

Reply:
Okay, attached the headers in both cases.